cyclient/scan_client.py

```diff
@@ -1,86 +1,86 @@
 import json
+from typing import List
+
 import requests.exceptions
 from requests import Response
-from typing import List
-from . import models
-from .cycode_token_based_client import CycodeTokenBasedClient
-from cli.zip_file import InMemoryZip
+
 from cli.exceptions.custom_exceptions import CycodeError, HttpUnauthorizedError
-from cyclient import logger
+from cli.zip_file import InMemoryZip
+from . import models
+from .cycode_client_base import CycodeClientBase
+from .scan_config.scan_config_base import ScanConfigBase
 
 
 class ScanClient:
-    SCAN_CONTROLLER_PATH = 'api/v1/scan'
-    SCAN_SERVICE_CONTROLLER_PATH = 'scans/api/v1/scan'
-    DETECTIONS_SERVICE_CONTROLLER_PATH = 'detections/api/v1/detections'
-
-    def __init__(self, client_id: str = None, client_secret: str = None):
-        self.cycode_client = CycodeTokenBasedClient(client_id, client_secret)
+    def __init__(self, scan_cycode_client: CycodeClientBase,
+                 scan_config: ScanConfigBase):
+        self.scan_cycode_client = scan_cycode_client
+        self.scan_config = scan_config
 
     def content_scan(self, scan_type: str, file_name: str, content: str, is_git_diff: bool = True):
-        path = f"{self.get_service_name(scan_type)}/{self.SCAN_CONTROLLER_PATH}/content"
+        path = self.scan_config.get_content_scan_controller_path(scan_type)
         body = {'name': file_name, 'content': content, 'is_git_diff': is_git_diff}
         try:
-            response = self.cycode_client.post(url_path=path, body=body)
+            response = self.scan_cycode_client.post(url_path=path, body=body)
             return self.parse_scan_response(response)
         except Exception as e:
             self._handle_exception(e)
 
     def file_scan(self, scan_type: str, path: str) -> models.ScanResult:
-        url_path = f"{self.get_service_name(scan_type)}/{self.SCAN_CONTROLLER_PATH}"
+        url_path = self.scan_config.get_scan_controller_path(scan_type)
         files = {'file': open(path, 'rb')}
         try:
-            response = self.cycode_client.post(url_path=url_path, files=files)
+            response = self.scan_cycode_client.post(url_path=url_path, files=files)
             return self.parse_scan_response(response)
         except Exception as e:
             self._handle_exception(e)
 
     def zipped_file_scan(self, scan_type: str, zip_file: InMemoryZip, scan_id: str, scan_parameters: dict,
                          is_git_diff: bool = False) -> models.ZippedFileScanResult:
-        url_path = f"{self.get_service_name(scan_type)}/{self.SCAN_CONTROLLER_PATH}/zipped-file"
+        url_path = self.scan_config.get_zipped_file_scan_controller_path(scan_type)
```

Comment:
get from the config the path to the controller and add the route and relevant query params here (relevant to all the requests here)

Reply:
Fixed